modules/seqexec/server/src/main/scala/seqexec/server/StepsView.scala

```diff
@@ -101,7 +103,7 @@ object StepsView {
 
   private def observeAction[F[_]](executions: List[ParallelActions[F]]): Option[Action[F]] =
     // FIXME This is too naive and doesn't work properly for N&S
-    executions.flatMap(_.toList).filter(_.kind === ActionType.Observe).headOption
+    executions.flatMap(_.toList).find(_.kind === ActionType.Observe)
```

Comment:
Why are we reverting this? If we do so, let's remove the `FIXME`

Reply:
This change does not address the `FIXME`. It is just a terser version of the same code.

Reply:
Your right